src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -29,98 +31,44 @@ def handle_media_info(
+            return False, None
     else:
-        bitstream_information = resource.get_bitstream_information(bitstream.value, permissions_lookup)
-    return success, bitstream_information
-
-
-def _handle_media_upload(
-    resource: XMLResource,
-    bitstream: XMLBitstream,
-    permissions_lookup: dict[str, Permissions],
-    sipi_server: Sipi,
-    imgdir: str,
-) -> BitstreamInfo | None:
-    """
-    Upload a bitstream file to SIPI
+        return True, BitstreamInfo(bitstream.value, bitstream.value, _permissions(bitstream, permissions_lookup))
 
-    Args:
-        resource: resource holding the bitstream
-        bitstream: the bitstream object
-        permissions_lookup: dictionary that contains the permission name as string and the corresponding Python object
-        sipi_server: server to upload
-        imgdir: directory of the file
 
-    Returns:
-        The information from sipi which is needed to establish a link from the resource
-    """
-    try:
-        resource_bitstream = _upload_bitstream(
-            resource=resource,
-            sipi_server=sipi_server,
-            imgdir=imgdir,
-            permissions_lookup=permissions_lookup,
-        )
-        msg = f"Uploaded file '{bitstream.value}'"
-        print(f"{datetime.now()}: {msg}")
-        logger.info(msg)
-        return resource_bitstream
-    except PermanentConnectionError as err:
-        msg = f"Unable to upload file '{bitstream.value}' of resource '{resource.label}' ({resource.res_id})"
-        print(f"{datetime.now()}: WARNING: {msg}: {err.message}")
-        logger.opt(exception=True).warning(msg)
+def _permissions(
+    bitstream: Optional[XMLBitstream], permissions_lookup: dict[str, Permissions]
+) -> Optional[Permissions]:
+    if bitstream is None:
         return None
-
-
-def _upload_bitstream(
-    resource: XMLResource,
-    sipi_server: Sipi,
-    imgdir: str,
-    permissions_lookup: dict[str, Permissions],
-) -> BitstreamInfo | None:
-    """
-    This function uploads a specified bitstream file to SIPI and then returns the file information from SIPI.
-
-    Args:
-        resource: resource that has a bitstream
-        sipi_server: server to upload
-        imgdir: directory of the file
-        permissions_lookup: dictionary that contains the permission name as string and the corresponding Python object
-
-    Returns:
-        The information from sipi which is needed to establish a link from the resource
-    """
-    if not resource.bitstream:
+    if bitstream.permissions is None:
         return None
-    img = sipi_server.upload_bitstream(Path(imgdir) / Path(resource.bitstream.value))
-    internal_file_name_bitstream = img["uploadedFiles"][0]["internalFilename"]
-    return resource.get_bitstream_information(
-        internal_file_name_bitstream=internal_file_name_bitstream,
-        permissions_lookup=permissions_lookup,
-    )
+    else:
+        return permissions_lookup.get(bitstream.permissions)
```

Comment:
It was not in the original code, but should we perhaps raise an error if the permissions don't exist as we do it in the `ResourceCreateClient`?

Reply:
I tried to keep the behaviour the same. AFAIU, the case of non existing permissions in the lookup is valid. Did I miss something?

Reply:
If the permissions string used in the xml is not defined we do not know what the user intended for the property. 

```
<text-prop name=":hasSimpleText">
            <text permissions="prop-unknown" encoding="utf8">blah</text>
</text-prop>
```

Permissions dict:

```
{ "prop-default": Permission }
```

Maybe it was a typo, so we do not quietly just ignore the invalid input, as we do here (because permissions are allowed to be `None` if nothing is specified), but we raise an user-facing error that communicates there is something wrong. I know this was not in the original, but if we see a small mistake such as this maybe we could fix it now. Or do a follow up PR.

Reply:
@Nora-Olivia-Ammann since you are the original author of the comment: could you create a ticket for this? I have already resolved the conversation.